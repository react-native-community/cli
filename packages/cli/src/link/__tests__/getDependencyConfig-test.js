/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+javascript_foundation
 */

const platforms = {
  ios: {
    dependencyConfig: () => ({ sampleiOSKey: '' }),
  },
  android: {
    dependencyConfig: () => ({ sampleAndroidKey: '' }),
  },
};

jest.setMock('../../core/getPackageConfiguration', folder => {
  if (folder === '/root/node_modules/abcd') {
    throw new Error('Cannot require');
  }
  return {};
});

const getDependencyConfig = require('../getDependencyConfig');

describe('getDependencyConfig', () => {
  it("should return an array of dependencies' config", () => {
    const dependencies = getDependencyConfig(
      { root: '/root' },
      platforms,
      'react-native-windows'
    );

    expect(dependencies).toMatchSnapshot();
  });

  it('should throw on invalid react-native dependency', () => {
    expect(() =>
      getDependencyConfig({ root: '/root' }, platforms, 'abcd')
    ).toThrowError();
  });
});
