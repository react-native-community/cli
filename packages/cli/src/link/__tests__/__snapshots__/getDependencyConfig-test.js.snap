// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDependencyConfig should return an array of dependencies' config 1`] = `
Object {
  "assets": Array [],
  "commands": Object {},
  "config": Object {
    "android": Object {
      "sampleAndroidKey": "",
    },
    "ios": Object {
      "sampleiOSKey": "",
    },
  },
  "name": "react-native-windows",
  "params": Array [],
  "path": "/root/node_modules/react-native-windows",
}
`;
